src/_pytest/_io/pprint.py

```diff
@@ -111,15 +111,15 @@ def _format(
             p(self, object, stream, indent, allowance, context, level + 1)
             context.remove(objid)
         elif (
-            _dataclasses.is_dataclass(object)  # type:ignore[unreachable]
+            _dataclasses.is_dataclass(object)
             and not isinstance(object, type)
-            and object.__dataclass_params__.repr
+            and object.__dataclass_params__.repr  # type:ignore[attr-defined]
```

Comment:
I assumed it was tested and is only mypy finding a new issue because the previous issue with unreachable was fixed so it hits new code.